span.go

```diff
@@ -119,9 +119,13 @@ func (s *spanImpl) FinishWithOptions(opts opentracing.FinishOptions) {
 	if s.tracer.Options.DebugAssertUseAfterFinish {
 		// This makes it much more likely to catch a panic on any subsequent
 		// operation since s.tracer is accessed on every call to `Lock`.
+		pool := s.tracer.spanPool
 		s.reset()
```

Comment:
this would arguably be easier to reason about if `reset()` did the `pool.Put(s)` internally... perhaps renaming `s/reset/resetAndAbandon/` or whatever.